includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,384 @@
+									$data['settings']
+								)
+							);
+						},
+						'permission_callback' => $can_authenticate,
+						'args'                => array(
+							'data' => array(
+								'type'       => 'object',
+								'required'   => true,
+								'properties' => array(
+									'settings' => array(
+										'type'      => 'object',
+										'required'  => true,
+										'questions' => array_fill_keys(
+											array_keys( static::$questions ),
+											array(
+												'type'  => 'array',
+												'items' => array( 'type' => 'string' ),
+											)
+										),
+									),
+								),
+							),
+						),
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Gets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|WP_Error User input settings.
+	 */
+	public function get_answers() {
+		$data = array(
+			'site' => $this->site_specific_questions->get(),
+			'user' => $this->user_specific_questions->get(),
+		);
+
+		$questions = static::$questions;
+		$settings  = array_merge( $data['site'], $data['user'] );
+
+		// If there are no settings, return default empty values.
+		if ( empty( $settings ) ) {
+			array_walk(
+				$questions,
+				function ( &$question ) {
+					$question['values'] = array();
+				}
+			);
+
+			return $questions;
+		}
+
+		$user_id = get_current_user_id();
+
+		foreach ( $settings as &$setting ) {
+			if ( ! isset( $setting['answeredBy'] ) ) {
+				continue;
+			}
+
+			$answered_by = intval( $setting['answeredBy'] );
+			unset( $setting['answeredBy'] );
+
+			if ( ! $answered_by || $answered_by === $user_id ) {
+				continue;
+			}
+
+			$setting['author'] = array(
+				'photo' => get_avatar_url( $answered_by ),
+				'name'  => get_the_author_meta( 'user_email', $answered_by ),
+			);
+		}
+
+		// If there are un-answered questions, return default empty values for them.
+		foreach ( $questions as $question_key => $question_value ) {
+			if ( ! isset( $settings[ $question_key ] ) ) {
+				$settings[ $question_key ]           = $question_value;
+				$settings[ $question_key ]['values'] = array();
+			}
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Determines whether the current user input settings have empty values or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The settings to check.
+	 * @return boolean|null TRUE if at least one of the settings has empty values, otherwise FALSE.
+	 */
+	public function are_settings_empty( $settings = array() ) {
+		if ( empty( $settings ) ) {
+			$settings = $this->get_answers();
+
+			if ( is_wp_error( $settings ) ) {
+				return null;
+			}
+		}
+
+		$empty_settings = array_filter(
+			$settings,
+			function( $setting ) {
+				return empty( $setting['values'] );
+			}
+		);
+
+		return 0 < count( $empty_settings );
+	}
+
+	/**
+	 * Sets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array|WP_Error User input settings.
+	 */
+	public function set_answers( $settings ) {
```

Comment:
We need to update the PHPDoc here as well.

Reply:
Updated, thank you!